packages/template-retail-react-app/app/components/search/index.jsx

```diff
@@ -116,18 +119,22 @@ const Search = (props) => {
         setSessionJSONItem(RECENT_SEARCH_KEY, searches)
     }
 
-    const debouncedSearch = debounce((input) => searchSuggestion.getSearchSuggestions(input), 300)
+    const debouncedSearch = debounce((input) => {
+        debouncedSearch.cancel()
```

Comment:
So I noticed a bug in the code where you would input a search value like "feeling", and if you were to delete the string letter by letter, once you got down to the threshold for making a search (I think it was 3 characters) it would invoke the search call because 300 ms has elapsed since last typing. Even tho your original intention was to delete the search string altogether. 

E.g. 

"feeling" (after 300 ms a search happens)
"feelin" (delete with a backspace)
"feeli" (delete with a backspace)
"feel" (delete with a backspace)
"fee" (delete with a backspace)
"fe" (delete with a backspace)
"f" (delete with a backspace)
"" (delete with a backspace)

In this scenario youd expect that no addition network requests are made, because the final search string is empty and less than the threshold. But you do end up seeing 1 additional request for "fee" because it was debounced and time elapsed and never cancelled.